Assets/MixedRealityToolkit-SDK/Features/UX/Interactable/Scripts/Layout/ButtonSize.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.UX
+{
+    [ExecuteInEditMode]
+    public class ButtonSize : MonoBehaviour
+    {
+        [Tooltip("A pixel to Unity unit conversion, Default: 2048x2048 pixels covers a 1x1 Unity Unit or default primitive size")]
+        public float BasePixelScale = 2048;
```

Comment:
Same as above

Reply:
Though "ButtonSize" might be the class where the single reference belongs, if it's not in the profiles.

Reply:
Thinking about this further, the component does not have to use a ButtonSize element as an Anchor, so without that requirement there is no way to get a BasePixelScale if it's not set on this object. It should just be 2048, it works best for designers building UI from redlines, but if someone wanted to throw it on something for a quick layout and didn't care about pixels, then they can set it to whatever works for them.

Reply:
This code is being updated in PR #2870.

Reply:
#2870 should go in first. I moved the code that was in #2852 (1 of 4) and moved it into #2870 (1.5 of 4), which is probably confusing. Each PR layers on top of the previous PRs because of their dependencies otherwise I can't test any changes in the later branches. I figured once one branch is through I would merge dev into the next branch is probably the cleanest path. It gets dicey though because any code changes (other than added comments) should get merged into my later branches for testing, but I'm worried about conflicts at some point.